.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,43 @@
+name: Build Documentation
+# Build on all branches
+on:
+  push:
+    branches:
+      - '*'
```

Comment:
only `master` or main maybe better?

Reply:
hi @yihong0618 thanks! this is setup to just build the docs on all pr's. this is ideal because if people make changes say to docstrings or if we end up adding examples to docstrings and use doctest to test them it will test the build before we merge and we can see if it passes. it does not update the docs in realtime it just builds them in ci. hope that makes sense.

Reply:
so in short it will test the docs for every PR. but it won't push them live. read the docs will update the docs when we merge to master/main